src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -36,17 +36,22 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
 
         private EnvDTE.SolutionEvents _solutionEvents;
         private CancellationTokenSource _workerCts;
-        private Lazy<Task> _backgroundJobRunner;
+        private AsyncLazy<bool> _backgroundJobRunner;
         private Lazy<BlockingCollection<SolutionRestoreRequest>> _pendingRequests;
         private BackgroundRestoreOperation _pendingRestore;
         private Task<bool> _activeRestoreTask;
         private int _initialized;
+        private TaskCompletionSource<bool> _solutionLoadedTcs;
```

Comment:
Since the `bool` result of this Task is not significant, consider using `AsyncManualResetEvent` instead, which is basically like this, but without claiming to return a result.
But as a bonus, you can *reset* it, so you don't have to keep creating new TaskCompletionSources, and worrying about making sure you complete the last before creating a new one.